apps/tests/lib/embedding_tests.ex

```diff
@@ -29,62 +41,80 @@ defmodule Tests.EmbeddingTests do
       {test_case, run_test(test_case)}
     end)
     |> Stream.each(&save_json_file/1)
-    |> Enum.count
+    |> Enum.count()
   end
 
   defp run_test(test_case) do
     TestSupervisor.terminate_all_children()
     {:ok, db_pid} = start_database(test_case)
     {:ok, index_pid} = start_index_server(test_case)
 
-    accuracy = test_embedding_model()
+    accuracy =
+      test_embedding_model(Map.get(test_case, :prepend_to_question), Map.get(test_case, :k))
+
     Logger.info("Test ended with success rate: #{Float.round(accuracy * 100, 2)}%")
 
     TestSupervisor.terminate_child(db_pid)
     TestSupervisor.terminate_child(index_pid)
     accuracy
   end
 
-  defp test_embedding_model() do
+  defp test_embedding_model(prepend, k) do
     all = DocumentationDatabase.get_all()
 
-    correct = all |> Stream.map(fn {id, fragment} ->
-      check_index(create_question(fragment), id)
-    end) |> Enum.count(fn result ->
-      result == :ok
-    end)
+    correct =
+      all

```

Comment:
IMO it's a good place to insert `Enum.take_random(@test_size)` to enable limiting the test size. Probably 1000 random test cases will be usually enough. We need to limit it when we test OpenAI embedding model as every test requires an API call and it becomes to lengthy.

Reply:
That's a good idea. I will also try to avoid fetching the entire database into memory though, as that might be painful for paging. Def will think of something